### PR TITLE
docs(contentful): contentful deployment

### PR DESCRIPTION
**Pull request by StaberindeZA**
Because:

* Contentful content model is created and deployed to Dev

This commit:

* Documents the deployment process for Contentful

Closes #FXA-7337

### CODE REVIEW

File: docs/tutorials/subscription-platform.md [hunk 1]
```diff
@@ -377,6 +378,72 @@ After the initial payment during subscription creation, the recurring payments f
...
+
+For a typical change and deployment, the following steps should be taken.
+
+1. Make changes in the `Dev` environment
+1. After a regular FxA release has been deployed to Stage, using the Merge App, merge changes from the `Dev` environtment alias to the `Stage` environment alias.
+1. Similarly, after a regular FxA release has been deployed to Production, using the Merge App, merge changes from the `Stage` environtment alias to the `Prod` environment alias.
+
+#### Sensitive or large content model changes
+
+In cases where many changes are made or a technically sensitive change that could introduce breaking changes, it is recommended to make the changes in a new environment, for example called `Feature`, before merging those changes to `Dev`.
```
Comment (sardesam) [inline docs/tutorials/subscription-platform.md:425]: - Reword - move "are made" to the end of the line before the comma.

> "In cases where many changes or a technically sensitive change that could introduce breaking changes **are made**..."

- "called" can be removed.
- Remove code block on "Dev" to be consistent with "Stage" and "Production" above on `L420` and `L421` respectively. Alternatively, wrap all in code blocks and change "Production" to "Prod."

File: docs/tutorials/subscription-platform.md [hunk 2]
```diff
@@ -377,6 +378,72 @@ After the initial payment during subscription creation, the recurring payments f
...
+The resulting deployment path would be as follows.
+
+```mermaid
+flowchart LR
+    Feature --> Dev --> Stage --> Prod
+```
+
+For a sensitive or large content model change and deployment, the following steps should be taken.
+
+1. (Optional) Delete the existing Feature environment. (Check with Subscription Platform team first before doing this)
```
Comment (sardesam) [inline docs/tutorials/subscription-platform.md:436]: - For consistency, "Feature" should be in a code block.
- Add period after "this" and before parenthesis.

File: docs/tutorials/subscription-platform.md [hunk 3]
```diff
@@ -377,6 +378,72 @@ After the initial payment during subscription creation, the recurring payments f
...
+```mermaid
+flowchart LR
+    Dev --> Stage --> Prod
+```
+
+For a typical change and deployment, the following steps should be taken.
+
+1. Make changes in the `Dev` environment
+1. After a regular FxA release has been deployed to Stage, using the Merge App, merge changes from the `Dev` environtment alias to the `Stage` environment alias.
+1. Similarly, after a regular FxA release has been deployed to Production, using the Merge App, merge changes from the `Stage` environtment alias to the `Prod` environment alias.
```
Comment (sardesam) [inline docs/tutorials/subscription-platform.md:421]: - Spelling error on both lines: "environtment."
- Change "Merge App" to "Merge app" as that is what Contentful does.

File: docs/tutorials/subscription-platform.md [hunk 4]
```diff
@@ -377,6 +378,72 @@ After the initial payment during subscription creation, the recurring payments f
...
+```
+
+For a sensitive or large content model change and deployment, the following steps should be taken.
+
+1. (Optional) Delete the existing Feature environment. (Check with Subscription Platform team first before doing this)
+1. Create new `Feature` environment from `Dev` alias environment
+1. Make changes in `Feature` environment
+1. Merge changes to `Dev` and follow typical deploymant path
+
+Once the changes in the `Feature` environment have been merged into `Dev`, the `Feature` environment can be deleted.
```
Comment (sardesam) [inline docs/tutorials/subscription-platform.md:441]: - Remove code block on "Dev" on both lines, to be consistent with "Stage" and "Production" above on `L420` and `L421` and other instance of "Dev" used in this context on `L425`. Alternatively, wrap all in code blocks and change "Production" to "Prod."
- Spelling error: "deploymant."

File: docs/tutorials/subscription-platform.md [hunk 5]
```diff
@@ -377,6 +378,72 @@ After the initial payment during subscription creation, the recurring payments f
...
+
+#### Hotfixes
+
+Hotfixes should follow either of the release paths discussed above, however mergers to the `Stage` and `Prod` do not need to follow the FxA release schedule.
```
Comment (sardesam) [inline docs/tutorials/subscription-platform.md:445]: - change "mergers" to "merges."
- "the" can be removed before "...`Stage` and `Prod`..."
- Similar to above (see other comments), remove code blocks on "Stage" and "Prod" and change "Prod" to "Production" for consistency. Alternatively, wrap all in code blocks and change all instances of "Production" to "Prod."

**Review by sardesam — CHANGES_REQUESTED** [review]
Excellent work. Looking good. Only a few tweaks. See my comments below and let me know if you have any questions!

File: docs/tutorials/subscription-platform.md (hunk 4 above)
Comment (StaberindeZA) [reply docs/tutorials/subscription-platform.md:441]: I reworded this a little so that it keeps the code block.

**Review by sardesam — APPROVED** [review]
Update looks great. I like how you reworded to specify environment and/or alias 👍 .

Approved!

**Pull request merged**